src/socket_base.cpp

```diff
@@ -546,7 +553,14 @@ int zmq::socket_base_t::bind (const char *endpoint_uri_)
         return rc;
     }
 
-    if (protocol == "pgm" || protocol == "epgm" || protocol == "norm") {
+    if (false
```

Comment:
I'm a bit concerned about these kind of ifdefs - when it resolves to false || false won't some compilers or static analyzers flag them as tauthologies?

Reply:
I am considering three other solution for this.

1. Make these protocol checking into inline function. ifdefs are handled in the function.
2. Split these condition checking codes for all ifdef cases
3. Do not cover ifdef for constants (protocol_name::pgm, epgm and norm) and keep previous ifdefs

What seems best solution for this problem?

Reply:
I'm fine with making it more verbose and split the condition for ifdefs - it's just 2 or 3

Reply:
Okay, I think second solution is better because it can improve conditional compile.

I'll fix and update.